### ISSUE TITLE
What do these various implementations mean?

### ISSUE THREAD

**Issue by 924973292** [MzeroMiko/VMamba#29]
What do these various implementations mean? 
# forward_core = forward_core_share_ssm
# forward_core = forward_core_share_a
# forward_core = forward_corev1
# forward_core = forward_corev2
# forward_core = forward_corev0
if  forward_core_share_ssm the orignal  s6 in mamba?

**Comment by MzeroMiko** [comment]
The `forward_corev0` is the original s6 in mamba, and all the checkpoints we provided are based on it.

**Comment by 924973292** [comment]
In forward_corev0, I wonder whether we can use the selective_scan_fn in the code you provided in the alter.. package?

**Comment by MzeroMiko** [comment]
Yes for now. In previous code, our `selective_scan_fn` is slightly different in inputs with `mamba_ssm` (we delete the `z`). But now, we use our own written  `‎SelectiveScan` which fits for either `selective_scan_cuda` or `selective_scan_cuda_core`.

We've tested in `check_vssm1_equals_vssm(forward_type="v0")`, that the code of new version acts the same with the previous one, so feel free to use it.
